crowbar_framework/app/models/backup.rb

```diff
@@ -23,6 +23,9 @@ class Backup < ActiveRecord::Base
   before_validation :save_or_create_archive, on: :create
   after_validation :delete_archive, unless: -> { errors.empty? }
   after_destroy :delete_archive
+  after_create do
```

Comment:
I think `after_create` is not good enough as we call `data` possibly at other times (not just when creating) -- for instance, when restoring.